packages/gateway/src/gateway-embedding-model.ts

```diff
@@ -0,0 +1,91 @@
+import type { EmbeddingModelV2 } from '@ai-sdk/provider';
+import {
+  combineHeaders,
+  createJsonResponseHandler,
+  postJsonToApi,
+  resolve,
+  type Resolvable,
+} from '@ai-sdk/provider-utils';
+import { z } from 'zod/v4';
+import type { GatewayConfig } from './gateway-config';
+import { asGatewayError } from './errors';
+import { parseAuthMethod } from './errors/parse-auth-method';
+
+export class GatewayEmbeddingModel implements EmbeddingModelV2<string> {
+  readonly specificationVersion = 'v2';
+  readonly supportedUrls = { '*/*': [/.*/] } as const;
+  readonly maxEmbeddingsPerCall = 2048;
```

Comment:
bedrock is an example where this is 1 -- let's test it and see how we behave, and consider vs. what we think a developer would want to happen.